instrumentation/nocode/src/main/java/com/splunk/opentelemetry/instrumentation/nocode/NocodeAttributesExtractor.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry.instrumentation.nocode;
+
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.incubator.semconv.code.CodeAttributesExtractor;
+import io.opentelemetry.instrumentation.api.incubator.semconv.util.ClassAndMethod;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import java.util.Collections;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+public class NocodeAttributesExtractor
+    implements AttributesExtractor<NocodeMethodInvocation, Void> {
+  private final AttributesExtractor<ClassAndMethod, Void> codeExtractor;
+
+  public NocodeAttributesExtractor() {
+    codeExtractor = CodeAttributesExtractor.create(ClassAndMethod.codeAttributesGetter());
+  }
+
+  @Override
+  public void onStart(
+      AttributesBuilder attributesBuilder, Context context, NocodeMethodInvocation mi) {
+    codeExtractor.onStart(attributesBuilder, context, mi.getClassAndMethod());
+
+    Map<String, String> attributes = Collections.EMPTY_MAP;
+    if (mi.getRule() != null) {
+      attributes = mi.getRule().attributes;
+    }
```

Comment:
There are a lot of opinions about direct field access, and yeah, with a lack of data classes available to us, I get the desire to not generate getters.....but this problem here would just disappear if it was encapsulated like`Map<String, String> attributes = methodInvocation.getRuleAttributes()`

Also reduces the law-of-demeter situation.